src/adapter/src/notice.rs

```diff
@@ -104,7 +105,13 @@ impl AdapterNotice {
             AdapterNotice::ClusterDoesNotExist { name: _ } => Some("Create the cluster with CREATE CLUSTER or pick an extant cluster with SET CLUSTER = name. List available clusters with SHOW CLUSTERS.".into()),
             AdapterNotice::DroppedActiveDatabase { name: _ } => Some("Choose a new active database by executing SET DATABASE = <name>.".into()),
             AdapterNotice::DroppedActiveCluster { name: _ } => Some("Choose a new active cluster by executing SET CLUSTER = <name>.".into()),
-            AdapterNotice::ClusterReplicaStatusChanged { status, .. } if *status == ClusterStatus::NotReady => Some("The cluster replica may be restarting or going offline.".into()),
+            AdapterNotice::ClusterReplicaStatusChanged { status, .. } => {
+                match status {
+                    ServiceStatus::NotReady(None) => Some("The cluster replica may be restarting or going offline.".into()),
+                    ServiceStatus::NotReady(Some(NotReadyReason::OOMKilled)) => Some("The cluster replica may have run out of memory and been killed.".into()),
```

Comment:
Why do you use "may" here? If k8s says the replica was OOM-killed, is there a chance that this is not correct?

Reply:
I don't know -- I doubt it! I was just trying to match the language in the other similar notice (the `NotReady(None)` case). Probably we should let @mjibson or someone else on Surfaces decide what the correct language is here for both cases.